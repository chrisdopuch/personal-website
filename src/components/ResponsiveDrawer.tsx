import AppBar from "@material-ui/core/AppBar";
import CssBaseline from "@material-ui/core/CssBaseline";
import Divider from "@material-ui/core/Divider";
import Drawer from "@material-ui/core/Drawer";
import Hidden from "@material-ui/core/Hidden";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { createStyles, Theme, withStyles, WithStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import MailIcon from "@material-ui/icons/Mail";
import MenuIcon from "@material-ui/icons/Menu";
import InboxIcon from "@material-ui/icons/MoveToInbox";
import React from "react";
import { Link, Route } from "react-router-dom";

const drawerWidth = 240;

const stylesDeclarations = (theme: Theme) =>
  createStyles({
    appBar: {
      marginLeft: drawerWidth,
      [theme.breakpoints.up("sm")]: {
        width: `calc(100% - ${drawerWidth}px)`,
      },
    },
    content: {
      flexGrow: 1,
      padding: theme.spacing.unit * 3,
    },
    drawer: {
      [theme.breakpoints.up("sm")]: {
        flexShrink: 0,
        width: drawerWidth,
      },
    },
    drawerPaper: {
      width: drawerWidth,
    },
    menuButton: {
      marginRight: 20,
      [theme.breakpoints.up("sm")]: {
        display: "none",
      },
    },
    root: {
      display: "flex",
    },
    toolbar: theme.mixins.toolbar,
  });

interface IResponsiveDrawerState {
  mobileOpen: boolean;
}

interface IResponsiveDrawerProps extends WithStyles<typeof stylesDeclarations> {
  title: string;
  theme: Theme;
}

class ResponsiveDrawer extends React.Component<IResponsiveDrawerProps, IResponsiveDrawerState> {
  public constructor(props: IResponsiveDrawerProps) {
    super(props);
    this.state = {
      mobileOpen: false,
    };
  }

  public handleDrawerToggle = () => {
    this.setState((state) => ({ mobileOpen: !state.mobileOpen }));
  };

  public render() {
    const { classes, theme, title } = this.props;

    const drawer = (
      <div>
        <div className={classes.toolbar} />
        <Divider />
        <List>
          {["Inbox", "Starred", "Send email", "Drafts"].map((text, index) => (
            <ListItem button={true} key={text}>
              <ListItemIcon>{index % 2 === 0 ? <InboxIcon /> : <MailIcon />}</ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>
        <Divider />
        <List>
          {["All mail", "Trash", "Spam"].map((text, index) => (
            <ListItem button={true} key={text}>
              <ListItemIcon>{index % 2 === 0 ? <InboxIcon /> : <MailIcon />}</ListItemIcon>
              <ListItemText primary={text} />
            </ListItem>
          ))}
        </List>
      </div>
    );

    return (
      <div className={classes.root}>
        <CssBaseline />
        <AppBar position="fixed" className={classes.appBar} title={title}>
          <Toolbar>
            <IconButton
              color="inherit"
              aria-label="Open drawer"
              onClick={this.handleDrawerToggle}
              className={classes.menuButton}
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" color="inherit" noWrap={true}>
              Responsive drawer
            </Typography>
          </Toolbar>
        </AppBar>
        <nav className={classes.drawer}>
          {/* The implementation can be swap with js to avoid SEO duplication of links. */}
          <Hidden smUp={true} implementation="css">
            <Drawer
              variant="temporary"
              anchor={theme.direction === "rtl" ? "right" : "left"}
              open={this.state.mobileOpen}
              onClose={this.handleDrawerToggle}
              classes={{
                paper: classes.drawerPaper,
              }}
              ModalProps={{
                keepMounted: true, // Better open performance on mobile.
              }}
            >
              {drawer}
            </Drawer>
          </Hidden>
          <Hidden xsDown={true} implementation="css">
            <Drawer
              classes={{
                paper: classes.drawerPaper,
              }}
              variant="permanent"
              open={true}
            >
              {drawer}
            </Drawer>
          </Hidden>
        </nav>
        <main className={classes.content}>
          <div className={classes.toolbar} />
          <Route path="/" exact={true} component={Index} />
          <Route path="/about/" component={About} />
          <Route path="/users/" component={Users} />
        </main>
      </div>
    );
  }
}

export default withStyles(stylesDeclarations, { withTheme: true })(ResponsiveDrawer);
