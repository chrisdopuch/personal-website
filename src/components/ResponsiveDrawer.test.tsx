import { createMuiTheme } from '@material-ui/core';
import { shallow } from 'enzyme';
import React from 'react';
import { ResponsiveDrawer } from './ResponsiveDrawer';

const appTheme = createMuiTheme({
  breakpoints: {
    up: jest.fn(),
  },
  spacing: {
    unit: 1,
  },
  typography: {
    useNextVariants: true,
  },
  zIndex: {
    drawer: 1,
  },
});

function getDefaultProps() {
  return {
    classes: {
      appBar: 'appBar',
      content: 'content',
      drawer: 'drawer',
      drawerPaper: 'drawerPaper',
      follow: 'follow',
      menuButton: 'menuButton',
      root: 'root',
      toolbar: 'toolbar',
    },
    theme: appTheme,
    title: 'test',
  };
}

describe('ResponsiveDrawer', () => {
  it('renders default props', () => {
    const wrapper = shallow(<ResponsiveDrawer {...getDefaultProps()} />);

    expect(wrapper).toMatchSnapshot();
  });
});
