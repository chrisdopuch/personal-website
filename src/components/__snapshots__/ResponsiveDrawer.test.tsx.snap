// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveDrawer renders default props 1`] = `
<div
  className="root"
>
  <WithStyles(AppBar)
    className="appBar"
    position="fixed"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Open drawer"
        className="menuButton"
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        test
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <nav
    className="drawer"
  >
    <Hidden
      implementation="css"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={true}
      smDown={false}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Drawer)
        ModalProps={
          Object {
            "keepMounted": true,
          }
        }
        anchor="left"
        classes={
          Object {
            "paper": "drawerPaper",
          }
        }
        onClose={[Function]}
        open={false}
        variant="temporary"
      >
        <div>
          <div
            className="toolbar"
          />
          <WithStyles(Divider) />
          <WithStyles(List)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Home"
              onClick={[Function]}
              to="/"
            >
              <WithStyles(ListItemIcon)>
                <pure(HomeIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Home"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="About me"
              onClick={[Function]}
              to="/about"
            >
              <WithStyles(ListItemIcon)>
                <pure(HelpIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="About me"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Gallery"
              onClick={[Function]}
              to="/gallery"
            >
              <WithStyles(ListItemIcon)>
                <pure(CameraIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Gallery"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Projects"
              onClick={[Function]}
              to="/projects"
            >
              <WithStyles(ListItemIcon)>
                <pure(CodeIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Projects"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
          <WithStyles(Divider) />
          <div
            className="social"
          >
            <div
              className="twitter"
            >
              <Follow
                onLoad={[Function]}
                options={Object {}}
                username="chrisdopuch"
              />
            </div>
            <a
              aria-label="Follow @chrisdopuch on GitHub"
              className="github-button"
              href="https://github.com/chrisdopuch"
            >
              Follow @chrisdopuch
            </a>
          </div>
        </div>
      </WithStyles(Drawer)>
    </Hidden>
    <Hidden
      implementation="css"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={false}
      smDown={true}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Drawer)
        classes={
          Object {
            "paper": "drawerPaper",
          }
        }
        open={true}
        variant="permanent"
      >
        <div>
          <div
            className="toolbar"
          />
          <WithStyles(Divider) />
          <WithStyles(List)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Home"
              onClick={[Function]}
              to="/"
            >
              <WithStyles(ListItemIcon)>
                <pure(HomeIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Home"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="About me"
              onClick={[Function]}
              to="/about"
            >
              <WithStyles(ListItemIcon)>
                <pure(HelpIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="About me"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Gallery"
              onClick={[Function]}
              to="/gallery"
            >
              <WithStyles(ListItemIcon)>
                <pure(CameraIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Gallery"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              component={[Function]}
              key="Projects"
              onClick={[Function]}
              to="/projects"
            >
              <WithStyles(ListItemIcon)>
                <pure(CodeIcon) />
              </WithStyles(ListItemIcon)>
              <WithStyles(ListItemText)
                primary="Projects"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
          <WithStyles(Divider) />
          <div
            className="social"
          >
            <div
              className="twitter"
            >
              <Follow
                onLoad={[Function]}
                options={Object {}}
                username="chrisdopuch"
              />
            </div>
            <a
              aria-label="Follow @chrisdopuch on GitHub"
              className="github-button"
              href="https://github.com/chrisdopuch"
            >
              Follow @chrisdopuch
            </a>
          </div>
        </div>
      </WithStyles(Drawer)>
    </Hidden>
  </nav>
  <main
    className="content"
  >
    <div
      className="toolbar"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/about"
    />
    <Route
      component={[Function]}
      path="/gallery"
    />
    <Route
      component={[Function]}
      path="/projects"
    />
  </main>
</div>
`;
